src/pinghelper.cpp

```diff
@@ -53,23 +53,28 @@ void PingHelper::start(const QString& serverIpv4Gateway,
   m_gateway = serverIpv4Gateway;
   m_source = deviceIpv4Address.section('/', 0, 0);
 
-  if (s_has_critical_ping_error) {
-    m_pingSender = new DummyPingSender(m_source, this);
-  } else {
-    m_pingSender =
+  m_pingSender =
 #if defined(MVPN_LINUX) || defined(MVPN_ANDROID)
-        new LinuxPingSender(m_source, this);
+      new LinuxPingSender(m_source, this);
 #elif defined(MVPN_MACOS) || defined(MVPN_IOS)
-        new MacOSPingSender(m_source, this);
+      new MacOSPingSender(m_source, this);
 #elif defined(MVPN_WINDOWS)
-        new WindowsPingSender(m_source, this);
+      new WindowsPingSender(m_source, this);
 #else
-        new DummyPingSender(m_source, this);
+      new DummyPingSender(m_source, this);
```

Comment:
Would you mind pulling these out in to a factory class?  I'm trying to clean up some of the uglier ifdefs for object creation.  You can look at ImageProviderFactory for what I did.
I'll approve anyway, but if you have time that would be great.

Reply:
Will do!